src/Atmos/Model/atmos_tendencies.jl

```diff
@@ -34,4 +34,15 @@ eq_tends(pv::PV, ::AtmosModel, ::Flux{FirstOrder}) where {PV <: Energy} =
 
 # TotalMoisture
 eq_tends(pv::PV, ::AtmosModel, ::Flux{FirstOrder}) where {PV <: TotalMoisture} =
```

Comment:
Perhaps sub-component types like `MoistureModel` should be added into these methods?

Reply:
I think we can do that, but the moment that `TotalMoisture` and `IceMoisture` or `LiquidMoisture` have different fluxes, they'll need their own methods. I'll combine them for now, and we can split them if we need.

Reply:
I've added `abstract type Moisture <: PrognosticVariable end`

Reply:
We will need them splitted very soon. Like this week - with the precipitation + 1-moment microphysics interface I am working on.

Would it make sense to merge that first so that you can see the more complicated example of moisture and precipitation sources?

Reply:
This is what I have for rain now - https://github.com/CliMA/ClimateMachine.jl/pull/1706 It has its own additional velocity that gets computed in the flux. 

I think it's mostly done. I got stuck adding precipitation to diagnostics. But should be done soon (I hope)